### DISCUSSION TITLE
Python scripts

### DISCUSSION THREAD

**Discussion by mdibit** [indomus/forum#33] [HUB: Home Assistant]
Prima di tutto un saluto agli iscritti a questo nuovo forum.
Ho sviluppato su Raspi4 un'applicazione in python per comunicare su linea seriale con un dispositivo che pilota una striscia LED (fatto da me). Pensavo di aggiungere a questa applicazione le funzioni per integrarla in Home Assistant ma, facendo dei test, sono giunto alla conclusione che non si capisce quali siano gli oggetti python o le librerie utilizzabili al di fuori del mondo Home Assistant.
Per il test ho dapprima creato uno script che riceve dati da HA e invia comandi per accendere luci gestite da HA, cioè senza uscire dal mondo HA. Tutto è OK, anche inserendo un istruzione "time.sleep" (tra l'altro senza eseguire il relativo import).   Appena inserisco una banale gestione di un file su disco non funziona più niente. Non ho nemmeno provato a eseguire un "import serial". Come si fa a sapere quali funzioni sono utilizzabili ?
Come va interpretata la nota : 
It is not possible to use Python imports with this integration. If you want to do more advanced scripts, you can take a look at AppDaemon or pyscript
Per avere completa libertà di sviluppo nello script bisogna passare a AppDaemon o generare una nuova integrazione?

**Comment by snakuzzo** [comment]
Non so quale sia la lista precisa di librerie incluse e disponibili per i python script di HA, ma di fatto non puoi fare import. Nel caso o fai come ti suggerisce la nota usando AppDaemon o pyscript...oppure puoi tranquillamente utilizzare degli script python veri e propri che richiami con uno shell command da HA...io ho fatto così per creare un paio di script per utenti sul vecchio forum e ne ho uno anche per me per generare dinamicamente un package ad hoc per l'integrazione con il decoder SkyQ

**Comment by mdibit** [comment]
Scusa il ritardo. Ti ringrazio per aver risvegliato un'idea che avevo
pensato come alternativa ma di cui mi sono poi
dimenticato, intestardendomi sulla soluzione della script
lanciata direttamente da HAss (sarebbe stata una soluzione molto elegante).
Funziona ottimamente, passando come argomenti nella linea di comando gli
stati delle 4 variabili che devo elaborare.
Soluzione molto flessibile.
Adesso svilupperò una command_line per andare a leggere lo stato


Il giorno ven 26 mar 2021 alle ore 20:45 Diego Ambrosanio <
***@***.***> ha scritto:
